test/execution/test_protocol.py

```diff
@@ -38,7 +38,7 @@ def test_trace_communication_actions_send():
     def protocol(alice, bob):
         tensor = alice.torch.tensor([1])
 
-        tensor.send(bob.worker)
+        tensor.send(bob)
```

Comment:
🎉 

What magic makes this work? 😄

Reply:
The magic is that I don't really send anything when I do `Placeholder.send()`. Just create a placeholder with `location` as `Role` and trace the action.